src/lib/enhanceElement.js

```diff
@@ -87,7 +87,7 @@ const enhanceWithPropTypes = _.curry((
       _.forEach(controlledPropUpdaterMap, (fn, key) => {
         const nextValue = this.props[key];
         if (nextValue !== prevProps[key]) {
-          fn(getInstanceFromComponent(this), nextValue);
+          fn.call(this, getInstanceFromComponent(this), nextValue);
```

Comment:
I'm seeing this `fn.call` as a bad sign, and I would say, put it in the third argument:

```js
fn(getInstanceFromComponent(this), nextValue, this);
```